CodeGeneration/Sources/generate-swiftsyntax/templates/swiftsyntax/SyntaxNodesFile.swift

```diff
@@ -304,3 +309,27 @@ private func generateSyntaxChildChoices(for child: Child) throws -> EnumDeclSynt
     }
   }
 }
+
+private func generateInitializerDocComment(for node: LayoutNode) -> SwiftSyntax.Trivia {
+  func generateParamDocComment(for child: Child) -> String? {
+    guard let documentation = child.documentation,
+      let firstLine = documentation.split(whereSeparator: \.isNewline).first
+    else {
+      return nil
+    }
+
+    return "   - \(child.varName): \(firstLine)"
+  }
+
+  let paramsDoc = docCommentTrivia(
+    from: ([
+      " - Parameters:",
+      "   - leadingTrivia: The leading ``Trivia`` for this node.",
+    ]
+      + node.children.compactMap(generateParamDocComment)
+      + ["   - trailingTrivia: The trailing ``Trivia`` for this node."])
+      .joined(separator: "\n")
+  )
```

Comment:
I’m not sure what you think but maybe using a multi-line string literal reads nicer here instead of the array.

```suggestion
  let paramsDoc = docCommentTrivia(
    from: """
       - Parameters:
         - leadingTrivia: The leading ``Trivia`` for this node.
      \(node.children.compactMap(generateParamDocComment).joined(separator: "\n"))
         - trailingTrivia: The trailing ``Trivia`` for this node.
       """
  )
```

Reply:
I love it! It's much more readable! But there is one issue with this code :/ If `node.children.compactMap(generateParamDocComment)` would return an empty array we would end with a comment like this:
```swift
///  - Parameters:
///    - leadingTrivia:  (...)
/// 
///    - trailingTrivia: (...)
```
the simplest way, I can think of, to address this issue in this approach is to add:
```swift
.split(whereSeparator: \.isNewline)
.filter { !$0.allSatisfy(\.isWhitespace) }
.joined(separator: "\n")
``` 
to this multi-line string. Or extract this code to `Utils` and make it reusable.

Reply:
Ah, I didn’t think about that. Extracting that logic to a function `removingEmptyLines` and just calling it is probably the easiest option

Reply:
I've found `Utils.swift` file in `Utils` module and it already contains very similar functions like: `dedented(string:)`, `flattened(indentedDocumentation:)`. Maybe I should stick to the pattern of these functions and decelerate the new one in this way:
```swift
/// Removes all empty lines from a multi-line string.
public func removedEmptyLines(string: String) -> String {
  string.split(whereSeparator: \.isNewline)
    .filter { !$0.allSatisfy(\.isWhitespace) }
    .joined(separator: "\n")
}
```

Reply:
That sounds like a good idea to me.